library/kani_core/src/models.rs

```diff
@@ -42,6 +44,71 @@ macro_rules! generate_models {
                     kani::kani_intrinsic()
                 }
             }
+
+            /// An offset model that checks UB.
+            #[kanitool::fn_marker = "OffsetModel"]
+            pub fn offset<T, P: Ptr<T>, O: ToISize>(ptr: P, offset: O) -> P {
+                let offset = offset.to_isize();
+                let t_size = core::mem::size_of::<T>() as isize;
+                if offset == 0 || t_size == 0 {
+                    // It's always safe to perform an offset of length 0.
+                    ptr
+                } else {
+                    let (byte_offset, overflow) = offset.overflowing_mul(t_size as isize);
```

Comment:
```suggestion
                    let (byte_offset, overflow) = offset.overflowing_mul(t_size);
```
You already cast to an `isize` when creating `t_size`, so I don't think this cast should be necessary.